src/server_manager/web_app/app.ts

```diff
@@ -653,23 +653,23 @@ export class App {
   }
 
   public async showServer(server: server.Server): Promise<void> {
-    const serverId = localServerId(server);
     this.selectedServer = server;
-    this.appRoot.selectedServerId = serverId;
-    localStorage.setItem(LAST_DISPLAYED_SERVER_STORAGE_KEY, serverId);
-
-    if (isManagedServer(server)) {
-      if (!(server as server.ManagedServer).isInstallCompleted()) {
-        this.appRoot.showProgress(this.makeDisplayName(server), true);
-        return;
-      }
-    }
+    this.appRoot.selectedServerId = localServerId(server);
+    this.setLastDisplayedServer(server);
     await this.updateServerView(server);
     this.appRoot.showServerView();
   }
 
+  setLastDisplayedServer(server: server.Server) {
+    localStorage.setItem(LAST_DISPLAYED_SERVER_STORAGE_KEY, localServerId(server));
+  }
+
   private async updateServerView(server: server.Server): Promise<void> {
-    if (await server.isHealthy()) {
+    if (isManagedServer(server) && !(server as server.ManagedServer).isInstallCompleted()) {
```

Comment:
This change should allow us to simplify the addServer a bit.
- I think the logic should be:
  1. updateServerView
  2. If installing, wait for install, then updateServerView again.
  3. Update server list entry
- We shouldn't need the `this.selectedServer === server` check anymore, because now a server being installed also uses the server view.

addServer is effectively where we manager the lifecycle of the server (the install/reachable state). Perhaps we can improve it in the future by leveraging the existing server state polling and emit domain events instead.

Reply:
Refactored